etna/search/views.py

```diff
@@ -283,6 +312,48 @@ def get_meta_title(self) -> str:
             title += ' for "' + query.replace('"', "'") + '"'
         return title
 
+    def get_datalayer_data(self, request: HttpRequest) -> Dict[str, Any]:
+        className = self.__class__.__name__
+        data["customDimension3"] = className
+        try:
+            group = self.get_api_filter_aggregations(self.form)
+            group = group[-1].replace("group:", "")
+        except Exception:
+            group = "none"
+        data["customDimension8"] = className.replace("SearchView", " Results: ") + group
+        if self.form.cleaned_data.get("q", ""):
+            data["customDimension9"] = self.form.cleaned_data.get("q", "")
+        else:
+            data["customDimension9"] = "*"
+        total_count = 0
+        try:
+            if group == "creator":
+                result = self.api_result["responses"][1]["aggregations"]["group"][
+                    "buckets"
+                ][0]["doc_count"]
+                total_count = result
+            elif className == "FeaturedSearchView":
+                for result in self.api_result["responses"]:
+                    total_count += result["hits"]["total"]["value"]
+            elif className == "WebsiteSearchView":
+                total_count = self.api_result["responses"][1]["hits"]["total"]["value"]
+            else:
+                result = self.api_result["responses"][1]["aggregations"][
+                    "catalogueSource"
+                ]["buckets"]
+                for bucket in result:
+                    total_count += bucket["doc_count"]
+            if total_count > 10000:
+                total_count = 10001
+        except Exception:
+            total_count = 0
+        data["customMetric1"] = total_count
+        try:
+            data["customMetric2"] = len(self.get_api_filter_aggregations(self.form)) - 1
+        except Exception:
+            data["customMetric2"] = 0
+        return data
```

Comment:
For `get_datalayer_data`  method, look into
implement some bits that are common like `customDimension3, customDimension9, customMetric2` by keeping with this method in this Base class.
For others which are specific to a view - `customDimension8, customMetric1` - `get_datalayer_data`  method in the respective view like CatalogueSearchView, WebsiteSearchView

Do we have `total_count` calculation elaborated somewhere to see?